src/util/codec/number.rs

```diff
@@ -110,6 +110,22 @@ pub trait NumberEncoder: Write {
         let u = order_encode_f64(f);
         self.encode_u64_desc(u)
     }
+
+    fn encode_u16_le(&mut self, v: u16) -> Result<()> {
+        self.write_u16::<LittleEndian>(v).map_err(From::from)
```

Comment:
There are many places that use `map_err`, using `try!` or `?` doesn't mean map_err should be forbidden.